server/src/main/java/org/candlepin/resource/GuestIdResource.java

```diff
@@ -167,38 +224,41 @@ public void updateGuest(
         @PathParam("consumer_uuid") @Verify(Consumer.class) String consumerUuid,
         @ApiParam("guest virtual uuid")
         @PathParam("guest_id") String guestId,
-        @ApiParam(name = "updated", required = true, value = "updated guest data to use") GuestId updated) {
+        @ApiParam(name = "updated", required = true, value = "updated guest data to use")
+        GuestIdDTO updatedDTO) {
 
         // I'm not sure this can happen
         if (guestId == null || guestId.isEmpty()) {
             throw new BadRequestException(
                 i18n.tr("Please supply a valid guest id"));
         }
 
-        if (updated == null) {
+        if (updatedDTO == null) {
             // If they're not sending attributes, we can get the guestId from the url
-            updated = new GuestId(guestId);
+            updatedDTO = new GuestIdDTO().setGuestId(guestId);
         }
 
         // Allow the id to be left out in this case, we can use the path param
-        if (updated.getGuestId() == null) {
-            updated.setGuestId(guestId);
+        if (updatedDTO.getGuestId() == null) {
+            updatedDTO.setGuestId(guestId);
         }
 
         // If the guest uuids do not match, something is wrong
-        if (!guestId.equalsIgnoreCase(updated.getGuestId())) {
+        if (!guestId.equalsIgnoreCase(updatedDTO.getGuestId())) {
             throw new BadRequestException(
                 i18n.tr("Guest ID in json \"{0}\" does not match path guest ID \"{1}\"",
-                    updated.getGuestId(), guestId));
+                    updatedDTO.getGuestId(), guestId));
         }
 
         Consumer consumer = consumerCurator.verifyAndLookupConsumer(consumerUuid);
-        updated.setConsumer(consumer);
+        GuestId guestIdEntity = new GuestId();
+        populateEntity(guestIdEntity, updatedDTO);
+        guestIdEntity.setConsumer(consumer);
         GuestId toUpdate = guestIdCurator.findByGuestIdAndOrg(guestId, consumer.getOwner());
```

Comment:
This could be done in such a way that we aren't creating a new entity. Lookup the actual DB entity, apply the changes directly to it, then merge. That way we don't have to juggle two instances of the entity and don't waste time allocating a new instance.

Reply:
done

Reply:
walking back on this one per our conversation online.

Reply:
ack

Reply:
walking back on this one per our conversation online.